### PR TITLE
[BugFix] Support BE log sys_log_level modify to take effect dynamically

### LINKED ISSUE #45062: [Bug] Support BE log 'sys_log_level' modify to take effect dynamically
### Steps to reproduce the behavior (Required)
dynamic parameters update not working for sys_log_level.
Execution does not take effect, update be sysLog:
curl -X POST http://{be_ip}:{be_http_port}/api/update_config?sys_log_level=ERROR

### Expected behavior (Required)
the sys log level should be updated to the specified log level

### Real behavior (Required)
the sys log level should be updated to the specified log level

### StarRocks version (Required)
main

### PR DESCRIPTION
**Pull request by rohitrs1983**
## Why I'm doing:
dynamic update for sys_log_level, sys log level should be updated to the specified log level

## What I'm doing:
dynamic update for sys_log_level, sys log level should be updated to the specified log level

Fixes #45062 

## What type of PR is this:

- [X] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [X] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [X] 3.3
  - [X] 3.2
  - [X] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/common/configbase.cpp [hunk 1]
```diff
@@ -229,9 +230,17 @@ Status set_config(const std::string& field, const std::string& value) {
     if (!it->second->set_value(value)) {
         return Status::InvalidArgument(fmt::format("Invalid value of config '{}': '{}'", field, value));
     }
+    update_config(field, value);
```
Comment (kevincai) [inline be/src/common/configbase.cpp:233]: what's the expected way to trigger this config update?

Comment (rohitrs1983) [reply be/src/common/configbase.cpp:233]: configure a dynamic parameter of a BE node by using the curl command.
curl -X POST http://{be_ip}:{be_http_port}/api/update_config?sys_log_level=ERROR

Comment (kevincai) [reply be/src/common/configbase.cpp:233]: check the `UpdateConfigAction::update_config` implementation, there is a update callback hook that processes the necessary update of configvar. shall keep the same way as other dynamic mutable vars.

**Review by kevincai — DISMISSED** [review]

**Comment by kevincai** [comment]
@rohitrs1983 thanks for the update, there is a clang-format check failure. please take a look and fix it as necessary.

**Review by sduzh — APPROVED** [review]

**Comment by sduzh** [comment]
@imay PTAL and check if you can skip the coverage check.

**Review by kevincai — APPROVED** [review]

**Comment by rohitrs1983** [comment]
> @imay PTAL and check if you can skip the coverage check.

@imay can u please merge this

**Review by imay — APPROVED** [review]

**Pull request merged**